snowplow_tracker/emitters.py

```diff
@@ -505,8 +516,8 @@ def flush(self) -> None:
         executes the flush method of the base Emitter class
         """
         with self.lock:
-            self.queue.put(self.buffer)
-            self.buffer = []
+            self.queue.put(self.event_store.event_buffer)
+            self.event_store.cleanup(self.event_store.event_buffer)
```

Comment:
I think this should just call `self.queue.put(self.event_store.get_events_batch())`. The `cleanup` is then called from the `send_events()` function.